### PR TITLE
[BPK-1261] Add eslint config to backpack-android

### PR DESCRIPTION
**Pull request by georgegillams**

### CODE REVIEW

File: .idea/modules.xml [hunk 1]
```diff
@@ -5,6 +5,7 @@
       <module fileurl="file://$PROJECT_DIR$/Backpack/Backpack.iml" filepath="$PROJECT_DIR$/Backpack/Backpack.iml" />
       <module fileurl="file://$PROJECT_DIR$/app/app.iml" filepath="$PROJECT_DIR$/app/app.iml" />
       <module fileurl="file://$PROJECT_DIR$/backpack-android.iml" filepath="$PROJECT_DIR$/backpack-android.iml" />
+      <module fileurl="file://$PROJECT_DIR$/backpack-android.iml" filepath="$PROJECT_DIR$/backpack-android.iml" />
```
Comment (matteo-hertel) [inline .idea/modules.xml]: any idea where this is from? is not different from the one above

File: .prettierignore [hunk 2]
```diff
@@ -0,0 +1 @@
+
```
Comment (matteo-hertel) [inline .prettierignore]: is an empty file needed?

File: package.json [hunk 3]
```diff
@@ -7,11 +7,19 @@
...
   "scripts": {
+    "precommit": "lint-staged",
     "preinstall": "node -e \"$(curl -fsSL https://raw.githubusercontent.com/Skyscanner/ensure-node-env/master/dist/index.js)\"",
-    "build": "gulp",
     "prerelease:env": ": ${BINTRAY_USER?Need to set BINTRAY_USER} ${BINTRAY_KEY?Need to set BINTRAY_KEY}",
     "prerelease": "npm run prerelease:env",
-    "release": "node release.js"
+    "release": "node release.js",
+    "test": "eslint . --ext .js,.jsx",
+    "build": "npm run test && gulp",
```
Comment (matteo-hertel) [inline package.json]: can we use npm's command hooks?
```
"prebuild": "npm run test",
"build": "gulp",
```

File: .idea/modules.xml (hunk 1 above)
Comment (shaundon) [reply .idea/modules.xml]: Yeah, no idea! I'll just remove it.

File: .prettierignore (hunk 2 above)
Comment (shaundon) [reply .prettierignore]: Yeah fair enough, will remove!

File: package.json (hunk 3 above)
Comment (shaundon) [reply package.json]: Sure - I didn't know about those

File: .idea/modules.xml (hunk 1 above)
Comment (shaundon) [reply .idea/modules.xml]: Actually shouldn't `.idea` just be gitignored?

File: package.json (hunk 3 above)
Comment (shaundon) [reply package.json]: Done

File: .prettierignore (hunk 2 above)
Comment (shaundon) [reply .prettierignore]: Done

File: .idea/modules.xml (hunk 1 above)
Comment (matteo-hertel) [reply .idea/modules.xml]: I tought so too, but the default android app has just some subfolders in the gitignore, not really sure what the deal it

**Pull request merged**